In [1]:
import pandas as pd
import numpy as np
pd.set_option('display.max_rows', 40)
pd.set_option('display.max_columns', 999)

In [2]:
dir_data= '../parquet_datasets'

In [3]:
medicos_ips=pd.read_parquet(f'{dir_data}/medicos_ips_servicios_corregido.parquet')

In [4]:
medicos_ips.head()

depa_nombre muni_nombre  habi_codigo_habilitacion  numero_sede  \
0   Antioquia    MEDELLÍN                 500100003            1   
1   Antioquia    MEDELLÍN                 500100008            1   
2   Antioquia    MEDELLÍN                 500100010            1   
3   Antioquia    MEDELLÍN                 500100012            1   
4   Antioquia    MEDELLÍN                 500100012            1   

                    sede_nombre  \
0   BERNARDO  AGUDELO JARAMILLO   
1     ANA CARMENZA POSADA ISAZA   
2  JUAN CAMILO RESTREPO SALCEDO   
3        INSTITUTO DEL CORAZON    
4        INSTITUTO DEL CORAZON    

                                           direccion telefono  \
0                         CL 19A N° 44- 25 Cons 2301  3665349   
1  CL  2 SUR # 46 - 55 CS 253 FASE II CLINICA LAS...  2687270   
2                        DIAG 75B # 2A - 80 CONS 809     None   
3                               CIRCULAR 73 N° 39-37  3207242   
4                               CIRCULAR 73 N° 39-37  3207242   

                                email   nits_nit dv clase_persona naju_codigo  \
0          bernardo.agudelo@gmail.com   71578430  0       NATURAL           1   
1      consultorioacposada@une.net.co   32530725  0       NATURAL           1   
2                                None   70569409  6       NATURAL           1   
3  calidad@institutodelcorazon.org.co  800107179  2      JURIDICO           1   
4  calidad@institutodelcorazon.org.co  800107179  2      JURIDICO           1   

  naju_nombre clpr_codigo                clpr_nombre ese nivel caracter  \
0     Privada           2  Profesional Independiente               None   
1     Privada           2  Profesional Independiente               None   
2     Privada           2  Profesional Independiente  NO           None   
3     Privada           1        Instituciones - IPS  NO           None   
4     Privada           1        Instituciones - IPS  NO           None   

  habilitado grse_codigo                                      grse_nombre  \
0         SI           3                                 Consulta Externa   
1         SI           3                                 Consulta Externa   
2         SI           3                                 Consulta Externa   
3         SI           7  Apoyo Diagnóstico y Complementación Terapéutica   
4         SI           7  Apoyo Diagnóstico y Complementación Terapéutica   

  serv_codigo                      serv_nombre ambulatorio hospitalario  \
0         320            320 -GINECOBSTETRICIA          SI           NO   
1         328            328 -MEDICINA GENERAL          SI           NO   
2         342                   342 -PEDIATRÍA          SI           NO   
3         701  701 -DIAGNÓSTICO CARDIOVASCULAR          SI           NO   
4         729        729 -TERAPIA RESPIRATORIA          SI           NO   

  unidad_movil domiciliario otras_extramural centro_referencia  \
0           NO           NO               NO                NO   
1           NO           NO               NO                NO   
2           NO           NO               NO                NO   
3           NO           NO               NO                SI   
4           NO           SI               NO                NO   

  institucion_remisora complejidad_baja complejidad_media complejidad_alta  \
0                   NO               NO                SI               NO   
1                   NO               SI                NO               NO   
2                   NO               NO                SI               NO   
3                   NO               NO                NO               SI   
4                   NO               SI                NO               NO   

  fecha_apertura fecha_cierre numero_distintivo numero_sede_principal  \
0       20110727          nan         DHS018733                    01   
1       20121121          nan         DHS069909                    01   
2       20030218          nan         DHS075097       

In [5]:
medicos_ips.columns

Index(['depa_nombre', 'muni_nombre', 'habi_codigo_habilitacion', 'numero_sede',
       'sede_nombre', 'direccion', 'telefono', 'email', 'nits_nit', 'dv',
       'clase_persona', 'naju_codigo', 'naju_nombre', 'clpr_codigo',
       'clpr_nombre', 'ese', 'nivel', 'caracter', 'habilitado', 'grse_codigo',
       'grse_nombre', 'serv_codigo', 'serv_nombre', 'ambulatorio',
       'hospitalario', 'unidad_movil', 'domiciliario', 'otras_extramural',
       'centro_referencia', 'institucion_remisora', 'complejidad_baja',
       'complejidad_media', 'complejidad_alta', 'fecha_apertura',
       'fecha_cierre', 'numero_distintivo', 'numero_sede_principal',
       'observaciones_serv_Res3100_2019', 'fecha_corte_REPS',
       'nombre_prestador', 'Unnamed: 41'],
      dtype='object')

In [6]:
medicos_ips['depa_nombre'].unique()

array(['Antioquia', 'Atlántico', 'Bogotá D.C', 'Bolívar', 'Boyacá',
       'Caldas', 'Caquetá', 'Cauca', 'Cesar', 'Córdoba', 'Cundinamarca',
       'Chocó', 'Huila', 'La Guajira', 'Magdalena', 'Meta', 'Nariño',
       'Norte de Santander', 'Quindío', 'Risaralda', 'Santander', 'Sucre',
       'Tolima', 'Valle del cauca', 'Arauca', 'Casanare', 'Putumayo',
       'San Andrés y Providencia', 'Amazonas', 'Guainía', 'Guaviare',
       'Vaupés', 'Vichada'], dtype=object)

In [7]:
# for i in medicos_ips['muni_nombre'].unique():
#     print(i)

In [8]:
list_unique=[]
for i in medicos_ips.iloc[:,10:23].columns:
    list_unique.append([i, medicos_ips[i].unique()])

**Toca corregir las columnas que se vallan a usar ya que muchas de las columnas tienen datos erroneos**

In [9]:
list_unique[1:9]

[['naju_codigo',
  array(['1', '4', 'JURIDICO', '3', '890985417', '1.0', '3.0', '4.0',
         '809011703', 'NATURAL', '7', '9'], dtype=object)],
 ['naju_nombre',
  array(['Privada', 'Pública', '4', 'Mixta', '1', '3', 'JURIDICO'],
        dtype=object)],
 ['clpr_codigo',
  array(['2', '1', '4', 'Pública', 'Privada', '3', 'JURIDICO', '1.0', '2.0',
         '4.0', '3.0'], dtype=object)],
 ['clpr_nombre',
  array(['Profesional Independiente', 'Instituciones - IPS',
         'Objeto Social Diferente a la Prestación de Servicios de Salud',
         '1', 'Transporte Especial de Pacientes', 'Privada', '4', '2'],
        dtype=object)],
 ['ese',
  array(['  ', 'NO', 'SI', 'Instituciones - IPS', 'Privada', '1',
         'Objeto Social Diferente a la Prestación de Servicios de Salud',
         'Profesional Independiente'], dtype=object)],
 ['nivel',
  array([' ', '3', '2', 'SI', '1', 'NO', '4', '  ', 'Instituciones - IPS'],
        dtype=object)],
 ['caracter',
  array([None, 'MUNICIPAL', '2', 

In [10]:
medicos_ips.iloc[:,10:18][medicos_ips['clpr_nombre'].isin(['1','4','2'])]

clase_persona naju_codigo naju_nombre clpr_codigo  \
3170                                   1    JURIDICO           4     Pública   
10523                                  2    JURIDICO           1     Privada   
15644    veronica.saldarriaga@uam.edu.co   890985417           3    JURIDICO   
15645    veronica.saldarriaga@uam.edu.co   890985417           3    JURIDICO   
15646    veronica.saldarriaga@uam.edu.co   890985417           3    JURIDICO   
...                                  ...         ...         ...         ...   
268931                                 9    JURIDICO           1     Privada   
268932                                 9    JURIDICO           1     Privada   
268933                                 9    JURIDICO           1     Privada   
274221                                 6     NATURAL           1     Privada   
274222                                 6     NATURAL           1     Privada   

       clpr_nombre                        ese nivel  \
3170             1        Instituciones - IPS    SI   
10523            1        Instituciones - IPS    NO   
15644            1                    Privada     4   
15645            1                    Privada     4   
15646            1                    Privada     4   
...            ...                        ...   ...   
268931           1        Instituciones - IPS         
268932           1        Instituciones - IPS         
268933           1        Instituciones - IPS         
274221           2  Profesional Independiente         
274222           2  Profesional Independiente         

                                                 caracter  
3170                                                    2  
10523                                                      
15644   Objeto Social Diferente a la Prestación de Ser...  
15645   Objeto Social Diferente a la Prestación de Ser...  
15646   Objeto Social Diferente a la Prestación de Ser...  
...                                                   ...  
268931                                                     
268932                                                     
268933                                                     
274221                                                     
274222                                                     

[622 rows x 8 columns]

**Existen registros con `clpr_nombre` erroneo y con el valor real en una columna diferente para 622 registros**.

In [11]:
clpr_nombre_valid=['Profesional Independiente', 'Instituciones - IPS','Objeto Social Diferente a la Prestación de Servicios de Salud','Transporte Especial de Pacientes']

In [12]:
clpr_nombre_cor_list=[]
for i in medicos_ips.index:
    if medicos_ips['clpr_nombre'].iloc[i] in clpr_nombre_valid:
        clpr_nombre_cor_list.append(medicos_ips['clpr_nombre'].iloc[i])
    elif medicos_ips['ese'].iloc[i]in clpr_nombre_valid:
        clpr_nombre_cor_list.append(medicos_ips['ese'].iloc[i])
    elif medicos_ips['nivel'].iloc[i] in clpr_nombre_valid:
        clpr_nombre_cor_list.append(medicos_ips['nivel'].iloc[i])
    elif medicos_ips['caracter'].iloc[i] in clpr_nombre_valid:
        clpr_nombre_cor_list.append(medicos_ips['caracter'].iloc[i])
    else:
        clpr_nombre_cor_list.append(np.nan)

In [13]:
clpr_nombre_cor_series=pd.Series(clpr_nombre_cor_list, dtype='str', index=medicos_ips.index)

In [14]:
medicos_ips.insert(15,'clpr_nombre_cor',clpr_nombre_cor_series)

In [15]:
medicos_ips['clpr_nombre_cor'].unique()

array(['Profesional Independiente', 'Instituciones - IPS',
       'Objeto Social Diferente a la Prestación de Servicios de Salud',
       'Transporte Especial de Pacientes'], dtype=object)

In [18]:
medicos_independientes=medicos_ips[medicos_ips['clpr_nombre_cor']=='Profesional Independiente']

In [24]:
medicos_independientes.shape

(81365, 42)

In [27]:
medicos_independientes[medicos_independientes['nombre_prestador'].isna()]

depa_nombre muni_nombre  habi_codigo_habilitacion  numero_sede  \
104041  Bogotá D.C      BOGOTÁ                1100136617            1   

                              sede_nombre             direccion    telefono  \
104041  ENRIQUE GUILLERMO TANG VILLANUEVA  KR 16 # 82 76 CS 317  3178326426   

                                  email    nits_nit   dv clase_persona  \
104041  enriquetangvillanueva@gmail.com  73201718.0  nan       NATURAL   

       naju_codigo naju_nombre clpr_codigo                clpr_nombre  \
104041         1.0     Privada         2.0  Profesional Independiente   

                  clpr_nombre_cor ese nivel caracter habilitado grse_codigo  \
104041  Profesional Independiente               None         SI         7.0   

                                            grse_nombre serv_codigo  \
104041  Apoyo Diagnóstico y Complementación Terapéutica       745.0   

                                       serv_nombre ambulatorio hospitalario  \
104041  745 -IMÁGENES DIAGNOSTICAS - NO IONIZANTES          .            .    

       unidad_movil domiciliario otras_extramural centro_referencia  \
104041           .            .                .                 .    

       institucion_remisora complejidad_baja complejidad_media  \
104041                   .                NO                SI   

       complejidad_alta fecha_apertura fecha_cierre numero_distintivo  \
104041               NO     20201006.0          nan        DHS1135029   

       numero_sede_principal       observaciones_serv_Res3100_2019  \
104041                    01  MODALIDADES: INTRAMURAL AMBULATORIO    

       fecha_corte_REPS nombre_prestador Unnamed: 41  
104041             None             None        None

In [30]:
(medicos_independientes['sede_nombre']==medicos_independientes['nombre_prestador']).sum()

76942